flutter-for-ios.md

```diff
@@ -27,3 +27,1759 @@ that are most relevant to your needs.
+the `Text` widget:
+
+<!-- skip -->
+{% prettify dart %}
+body: new Center(
+  child: new TextField(
+    decoration: new InputDecoration(hintText: "This is a hint"),
+  ),
+)
+{% endprettify %}
+
+## How do I show validation errors?
+
+Just as you would with a "hint", you can pass in an `InputDecoration` object
+to the decoration constructor for the `Text` widget.
+
+However, you would not want to start off with showing an error and typically
+would want to show it when the user has entered some invalid data. This can be
+done by updating the state and passing in a new `InputDecoration` object.
+
+<!-- skip -->
+{% prettify dart %}
+class SampleApp extends StatelessWidget {
+  // This widget is the root of your application.
+  @override
+  Widget build(BuildContext context) {
+    return new MaterialApp(
+      title: 'Sample App',
+      theme: new ThemeData(
+        primarySwatch: Colors.blue,
+      ),
+      home: new SampleAppPage(),
+    );
+  }
+}
+
+class SampleAppPage extends StatefulWidget {
+  SampleAppPage({Key key}) : super(key: key);
+
+  @override
+  _SampleAppPageState createState() => new _SampleAppPageState();
+}
+
+class _SampleAppPageState extends State<SampleAppPage> {
+  String _errorText;
+
+  @override
+  Widget build(BuildContext context) {
+    return new Scaffold(
+      appBar: new AppBar(
+        title: new Text("Sample App"),
+      ),
+      body: new Center(
+        child: new TextField(
+          onSubmitted: (String text) {
+            setState(() {
+              if (!isEmail(text)) {
+                _errorText = 'Error: This is not an email';
+              } else {
+                _errorText = null;
+              }
+            });
+          },
+          decoration: new InputDecoration(hintText: "This is a hint", errorText: _getErrorText()),
+        ),
+      ),
+    );
+  }
+
+  _getErrorText() {
+    return _errorText;
+  }
+
+  bool isEmail(String em) {
+    String emailRegexp =
+        r'^(([^<>()[\]\\.,;:\s@\"]+(\.[^<>()[\]\\.,;:\s@\"]+)*)|(\".+\"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$';
+
+    RegExp regExp = new RegExp(p);
+
+    return regExp.hasMatch(em);
+  }
+}
+{% endprettify %}
+
+# Interacting with hardware, third party services and the platform
+
+## How do I access the GPS sensor?
+
+To access the GPS sensor you can use the [`location`](https://pub.dartlang.org/packages/location)
+community plugin.
+
+## How do I access the Camera?
+
+A popular plugin to access the camera is [`image_picker`](https://pub.dartlang.org/packages/image_picker).
+
+## How do I Log in with Facebook?
+
+To Log in with Facebook, you can use the
+[`flutter_facebook_login`](https://pub.dartlang.org/packages/flutter_facebook_login) community plugin.
+
+## How do I use Firebase features?
+
+Most Firebase functions are covered by [first or third party plugins](https://pub.dartlang.org/flutter/packages?q=firebase).
```

Comment:
Please reference the official 1st party plug-ins here

Reply:
Done in #993